src/components/Personalization/createExecuteDecisions.js

```diff
@@ -20,6 +20,31 @@ const buildActions = decision => {
   return decision.items.map(item => assign({}, item.data, { meta }));
 };
 
+const dedupe = metas => {
+  const result = [];
+  const set = new Set();
+
+  metas.forEach(item => {
+    if (set.has(item.id)) {
+      return;
+    }
+
+    set.add(item.id);
+    result.push(item);
+  });
+
+  return result;
+};
```

Comment:
This should probably be extracted to a utility. Here's my thought:

```
// identity.js
const identity = value => value;

// dedupe.js
const dedupe = (items, keyResolver = identity) => {
  const set = new Set();
  return items.filter(item => {
    const key = keyResolver(item);
    if (set.has(key)) {
      return false;
    }

    set.add(key);
    return true;
  });
};
```

Example usage:

```
const dedupedMetas = dedupe(metas, meta => meta.id);
```